Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -0,0 +1,645 @@
+    updater.template connect<&voidKalmanUpdater<traj_t>>();
+    outlierFinder.template connect<&voidOutlierFinder<traj_t>>();
+  }
+};
+
+/// Combined options for the Global-Chi-Square fitter.
+///
+/// @tparam traj_t The trajectory type
+template <typename traj_t>
+struct Gx2FitterOptions {
+  /// PropagatorOptions with context.
+  ///
+  /// @param gctx The geometry context for this fit
+  /// @param mctx The magnetic context for this fit
+  /// @param cctx The calibration context for this fit
+  /// @param extensions_ The KF extensions
+  /// @param pOptions The plain propagator options
+  /// @param rSurface The reference surface for the fit to be expressed at
+  /// @param mScattering Whether to include multiple scattering
+  /// @param eLoss Whether to include energy loss
+  /// @param freeToBoundCorrection_ Correction for non-linearity effect during transform from free to bound
+  /// @param nUpdateMax_ Max number of iterations for updating the parameters
+  Gx2FitterOptions(const GeometryContext& gctx,
+                   const MagneticFieldContext& mctx,
+                   std::reference_wrapper<const CalibrationContext> cctx,
+                   Gx2FitterExtensions<traj_t> extensions_,
+                   const PropagatorPlainOptions& pOptions,
+                   const Surface* rSurface = nullptr, bool mScattering = false,
+                   bool eLoss = false,
+                   const FreeToBoundCorrection& freeToBoundCorrection_ =
+                       FreeToBoundCorrection(false),
+                   const size_t nUpdateMax_ = 5)
+      : geoContext(gctx),
+        magFieldContext(mctx),
+        calibrationContext(cctx),
+        extensions(extensions_),
+        propagatorPlainOptions(pOptions),
+        referenceSurface(rSurface),
+        multipleScattering(mScattering),
+        energyLoss(eLoss),
+        freeToBoundCorrection(freeToBoundCorrection_),
+        nUpdateMax(nUpdateMax_) {}
+
+  /// Contexts are required and the options must not be default-constructible.
+  Gx2FitterOptions() = delete;
+
+  /// Context object for the geometry
+  std::reference_wrapper<const GeometryContext> geoContext;
+  /// Context object for the magnetic field
+  std::reference_wrapper<const MagneticFieldContext> magFieldContext;
+  /// context object for the calibration
+  std::reference_wrapper<const CalibrationContext> calibrationContext;
+
+  Gx2FitterExtensions<traj_t> extensions;
+
+  /// The trivial propagator options
+  PropagatorPlainOptions propagatorPlainOptions;
+
+  /// The reference Surface
+  const Surface* referenceSurface = nullptr;
+
+  /// Whether to consider multiple scattering
+  bool multipleScattering = false;
+
+  /// Whether to consider energy loss
+  bool energyLoss = false;
+
+  /// Whether to include non-linear correction during global to local
+  /// transformation
+  FreeToBoundCorrection freeToBoundCorrection;
+
+  /// Max number of iterations during the fit
+  const size_t nUpdateMax = 5;
```

Comment:
I generally would advice not to make members const, since this inhibits a bunch of compiler generated code (no copy assignment, no move constrution...)